lib/blockchain/chaindb.js

```diff
@@ -64,6 +64,10 @@ class ChainDB {
 
     this.cacheHash = new LRU(this.options.entryCache, null, BufferMap);
     this.cacheHeight = new LRU(this.options.entryCache);
+
+    // Keeps a copy of all NameStates in memory
+    // for their entire auction cycle.
+    this.nameCache = new LRU(this.options.nameCache, null, BufferMap);
```

Comment:
With current setup control of the nameCache is slightly weird. If user sets name-cache to, lets say, 5k - it will evict items when this number is reached. But user can't actually guess the memory which will be taken by this cache and this may be more flexible if we instead measure this thing using bytes instead of number of items.

5k in the worst case scenario will store: 5000 * 668 bytes = 3.34 MB, but of course most of the namestates are not of that size, that's the worst case.

https://github.com/handshake-org/hsd/blob/484ede0b45c92249a4af81faf6579dcf1611edfc/lib/covenants/namestate.js#L850-L864

Of course this is not critically important, as we can play with numbers and figure out some optimal number for IDB. I am planning to do some tests on this front and figure out Number of reads/writes that occur for the whole sync and have better numbers, but that will have to wait.

We could alternative use size instead and measure limits in bytes:
```js
this.nameCache = new LRU(size, (v, k) => v.length + k.length, BufferMap);
```

Reply:
> this.nameCache = new LRU(size, (v, k) => v.length + k.length, BufferMap);


Oh thanks for this, super interesting yeah I saw that in the LRU constructor but now I understand.